contracts/RestrictedNORI.sol

```diff
@@ -0,0 +1,1024 @@
+    uint256[] memory quantitiesToBurnForHolders = new uint256[](
+      tokenHolders.length()
+    );
+
+    uint256 cumulativeQuantityToBurn = 0;
+    for (uint256 i = 0; i < (tokenHolders.length() - 1); i++) {
+      uint256 quantityToBurnForHolder = _quantityToRevokePerTokenHolder(
+        quantityToRevoke,
+        scheduleId,
+        tokenHolders.at(i)
+      );
+      quantitiesToBurnForHolders[i] = quantityToBurnForHolder;
+      cumulativeQuantityToBurn += quantityToBurnForHolder;
+    }
+    quantitiesToBurnForHolders[tokenHolders.length() - 1] =
+      quantityToRevoke -
+      cumulativeQuantityToBurn;
+
+    for (uint256 i = 0; i < (tokenHolders.length()); i++) {
+      super._burn(
+        tokenHolders.at(i),
+        scheduleId,
+        quantitiesToBurnForHolders[i]
+      );
+      schedule.quantitiesRevokedByAddress[
+        tokenHolders.at(i)
+      ] += quantitiesToBurnForHolders[i];
+    }
+
+    schedule.totalQuantityRevoked += quantityToRevoke;
+    _bridgedPolygonNori.send(
+      // solhint-disable-previous-line check-send-result, because this isn't a solidity send
+      to,
+      quantityToRevoke,
+      ""
+    );
+    emit TokensRevoked(
+      block.timestamp, // solhint-disable-line not-rely-on-time, this is time-dependent
+      removalId,
+      scheduleId,
+      quantityToRevoke
+    );
+  }
+
+  /**
+   * Hook that is called before any token transfer. This includes minting and burning, as well as batched variants.
+   *
+   * @dev Follows the rules of hooks defined [here](
+   *  https://docs.openzeppelin.com/contracts/4.x/extending-contracts#rules_of_hooks)
+   * @dev See the ERC1155 specific version [here](
+   *  https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#ERC1155-_beforeTokenTransfer-address-address-address-uint256---uint256---bytes-)
+   *
+   * ##### Requirements:
+   *
+   * - the contract must not be paused
+   * - One of the following must be true:
+   *    - the operation is a mint (which should ONLY occur when BridgedPolygonNORI is being wrapped via `_depositFor`)
+   *    - the operation is a burn, which only happens during revocation and withdrawal:
+   *      - if the operation is a revocation, that permission is enforced by the TOKEN_REVOKER_ROLE
+   *      - if the operation is a withdrawal the burn amount must be <= the sender's claimable balance
+   *    - the operation is a transfer and _all_ of the following must be true:
+   *      - the operator is operating on their own balance (enforced in the inherited contract)
+   *      - the operator has sufficient balance to transfer (enforced in the inherited contract)
+   */
+  function _beforeTokenTransfer(
+    address operator,
+    address from,
+    address to,
+    uint256[] memory ids,
+    uint256[] memory amounts,
+    bytes memory data
+  ) internal override(ERC1155SupplyUpgradeable, ERC1155PausableUpgradeable) {
+    bool isBurning = to == address(0);
+    bool isWithdrawing = isBurning && from == operator;
```

Comment:
Is this true? Are we disabling operator transfers?  What conditions would see this false when isBurning is true?

Reply:
we could discuss this further but jaycen and I were thinking not to disable operator transfers since we looked at the code again and realized it wasn't as permissive as you and I thought it was during the review meeting for this